src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -0,0 +1,180 @@
+        """Returns the adjacency matrix of the graph."""
+        return self._adjacency
+
+    def get_complement_graph(self) -> netx.Graph:
+        """Returns the complement graph in networkx format."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        c_graph = self._get_graph_from_adjacency_matrix(c_adjacency)
+        return c_graph
+
+    def get_complement_graph_matrix(self) -> np.ndarray:
+        """Returns the adjacency matrix of the complement graph."""
+        c_adjacency = self._get_adjacency_of_complement_graph(self._adjacency)
+        return c_adjacency
+
+    def get_as_qubo(self, w_diag: float, w_off: float) -> QUBO:
+        """
+        Creates a QUBO whose solution corresponds to the maximum independent
+        set (MIS) of the graph defined by the input adjacency matrix.
+
+        The goal of the QUBO is to minimize the cost
+            min x^T * Q * x ,
+        where the vector x is defined as:
+            x_i = 1 if vertex i is part of the MIs
+            x_i = 0 if vertex i is not part of the MIS,
+        and the QUBO matrix is given by
+            Q_ii = w_diag
+            Q_ij = w_off (for i~=j) .
+        """
+        q = self._get_qubo_cost_from_adjacency(self._adjacency, w_diag, w_off)
+        qubo = QUBO(q)
+        return qubo
+
+    def _get_random_graph_mtx(self) -> np.ndarray:
+        """
+        Creates an undirected graph with random connectivity between nodes
+        and returns its adjacency matrix.
+        """
+
+        np.random.seed(self.seed)
+
+        random_matrix = np.random.rand(self.num_vertices, self.num_vertices)
+        adjacency = (random_matrix < self.connection_prob).astype(int)
+
+        adjacency = np.triu(adjacency)
+        # ensure that the matrix is symmetric
+        adjacency += adjacency.T - 2 * np.diag(adjacency.diagonal())
+
+        return adjacency
+
+
+def find_maximum_independent_set(mis_problem: MISProblem) -> \
+        npt.ArrayLike:
+    """
+    Find and return the maximum independent set of a graph based on its
+    adjacency matrix.
+
+    Uses Networkx to solve the equivalent maximum clique problem.
+    Get a graph whose maximum clique corresponds to the maximum independent
+    set of that defined by the input connectivity matrix.
+
+    The  maximum independent set for the graph G={V,E} is equivalent to the
+    maximum-clique of the graph H={V,E_c}, where E_c is the complement of E.
+
+    Returns
+    -------
+    solution:  Array[binary]
+    Vector of length equal to the number of vertices in the graph.
+    The ith entry of the vector determines if the ith vertex is a
+    member of the MIS.
+    """
+    c_graph = mis_problem.get_complement_graph()
+    maximum_clique = find_maximum_clique(undirected_graph=c_graph)
+    mis = indices_to_binary_vector(maximum_clique, mis_problem.num_vertices)
+    return mis
+
+
+def find_maximum_clique(undirected_graph: netx.Graph):

```

Comment:
I would advise against methods that just call another method.
Instead, why don't we just move this single line into the find_maximum_independent_set method?

Reply:
Agree.